### PR TITLE
Add VCX Java demo

### PR DESCRIPTION
**Pull request by baegjae**
This VCX Java Alice/Faber demo was created with reference to the VCX node demo.

Also, it contains small bug fix in VCX Java wrapper APIs.
- issuerSendCredentialCB - Remove credentialDefId parameter (incurred error)
- vcxProofGetStateCB - Remove proofHandle parameter (returned wrong state)

### CODE REVIEW

File: vcx/wrappers/java/src/main/java/com/evernym/sdk/vcx/issuer/IssuerApi.java [hunk 1]
```diff
@@ -155,20 +155,20 @@ public void callback(int commandHandle, int err, int state) {
     }
     private static Callback issuerSendCredentialCB = new Callback() {
         @SuppressWarnings({"unused", "unchecked"})
-        public void callback(int commandHandle, int err, String credentialDefId) {
-            logger.debug("callback() called with: commandHandle = [" + commandHandle + "], err = [" + err + "], credentialDefId = [" + credentialDefId + "]");
-            CompletableFuture<String> future = (CompletableFuture<String>) removeFuture(commandHandle);
+        public void callback(int commandHandle, int err) {
```
Comment (Patrik-Stas) [inline vcx/wrappers/java/src/main/java/com/evernym/sdk/vcx/issuer/IssuerApi.java:158]: Note for other reviewers: this seems correct, see the callback definition here
https://github.com/hyperledger/indy-sdk/blob/master/vcx/libvcx/src/api/issuer_credential.rs#L196

File: vcx/wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java [hunk 2]
```diff
@@ -123,8 +123,8 @@ public void callback(int commandHandle, int err, String proofRequests) {
 
     private static Callback vcxProofGetStateCB = new Callback() {
         @SuppressWarnings({"unused", "unchecked"})
-        public void callback(int commandHandle, int err, int proofHandle, int state) {
-            logger.debug("callback() called with: commandHandle = [" + commandHandle + "], err = [" + err + "], proofHandle = [" + proofHandle + "], state = [" + state + "]");
+        public void callback(int commandHandle, int err, int state) {
```
Comment (Patrik-Stas) [inline vcx/wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java:126]: Note for other reviewers: this seems correct, see the callback definition here
https://github.com/hyperledger/indy-sdk/blob/master/vcx/libvcx/src/api/disclosed_proof.rs#L438

**Comment by Patrik-Stas** [comment]
Looks good to me 👍 Let's wait for @jovfer 's review as well.

**Comment by baegjae** [comment]
@jovfer Currently, demo is not executed when using maven's `com.evernym:vcx:0.8.0`, so it is written to use the local vcx java wrapper (jar) temporarily. In my opinion, it would be better to merge this PR first. Next, as soon as this PR's bugfix applies to maven, I will register a PR that imports the vcx java wrapper from maven.

**Comment by Patrik-Stas** [comment]
@baegjae I see it's common practice to commit gradlew files cvs, and the .jar file is just 50kb, but for sake of security, can you make binary file checksum as part of jenkins pipeline? 
See: https://docs.gradle.org/current/userguide/gradle_wrapper.html#wrapper_checksum_verification

**Comment by baegjae** [comment]
@Patrik-Stas I applied the your request. If you have any further needs, please let me know.

**Comment by Patrik-Stas** [comment]
@baegjae Thanks for update - however still missing the check part. I'd rather not have the checksum file as part of the repo - instead either in jenkins, or as a github action, I'd like to:
- download the checksum file for this gradlew version from `gradle.org`
- verify the checksum as part of the pipeline

I think actually this will be much easier to add as github action. Extending existing Jenkins pipeline can be quite difficult due to extremely long feedback loop.

**Comment by kukgini** [comment]
Since VCX is forked and will be removed here, I think this PR should be dropped in this repository. https://github.com/hyperledger/indy-sdk/pull/2416

**Pull request closed without merging**